DEVX-4428: Vehicle Dispatch

oliverfcarson opened the pull request:
Please review the commit details for detailed information. Please note two things:

1. I did not incorporate the stretch goal yet - it will take some rework, but is potentially doable. Let me know if this is a priority that should be implemented.
2. There is an outdated library that is being used, [google.maps.marker](https://developers.google.com/maps/documentation/javascript/advanced-markers/migration). I tried to update the API based on the migration instructions, but keep running into a lot of issues, as I am quite unfamiliar with the Maps API. Although the legacy library is outdated, is is currently still supported (just throwing a warning in the console). Let me know if this is also something that needs to be fixed before published.

oliverfcarson committed:
Initial Check-In

-Initial set-up of transport and delivery demo to work in showcase demo.
-Removal of pubnub site demo functionality.
-Clean up, initialization of pubnub object, and changes to be made in following commits.

oliverfcarson committed:
Checkpoint - Add footer, Map and Dispatch Vehicles sizing

-Updated with showcase footer.
-Map now fills in most of the screen.
-Only the "dispatch vehicles" section will scroll when appropriate.

oliverfcarson committed:
Utilize shared PubNub Instance, code cleanup

-Utilize shared pubnub instance across healthcare showcase
-Removal of pubnub_setup.js file, no longer necessary for this app.
-Moved starting app functionality from pubnub_handler to vehicle-dispatch.js. Removed pubnub_handler.js file.
-Some code and comment cleanup.

oliverfcarson committed:
Styling on Mobile vs Desktop

-Styling updates on mobile and desktop so the bottom nav bar always appears on the bottom, regardless of device.

oliverfcarson committed:
Update Styling, Healthcare Terminology, Removal of Unnecessary Features

-Updated styling.
-Update terminology of vehicle names.
-Sensor data (bpm) is now using a new formula to calculate and simulate heart rate
-Uploaded ambulance svg to read from when displaying icon due to loading file issues.

oliverfcarson committed:
Icon Image Updates, Healthcare Updates

-Removal of local ambulance.svg. Uploaded to Contentful Media Servers (to avoid CORS issues).
-Removal of presence icon and logic powering.
-Removal of Reboot Senor UI and background logic.
-Removal of "Remaining Deliveries" in vehicle information.
-Changed "Type" to "Patient Monitoring Type" for the vehicle description information
-Changed "Next Delivery" to "Next Stop" for the vehicle description information
-Changed "Change the Next Delivery" text when editing the vehicle modal to "Change the Next Stop".
-Code cleanup (spacing, unnecessary comments, etc).

oliverfcarson committed:
Presence Cleanup

-Clean up old presence handling functions since we longer need presence updates.

oliverfcarson committed:
Old Presence Code Removal

-removal of old presence code

darryncampbell-pubnub commented:
Hey Oliver, I only see two issues:

1. There is a missing graphic at /vehicle-dispatch/pn_small.png, so if you send a push message to the driver, the image is missing
2. There is no actionCompleted call for the `Launch the Vehicle Dispatch app` feature.

Other than that, looks good

oliverfcarson committed:
PR Fixes & actionCompleted fixes

-Fixed issue of missing graphic when sending a push message to the driver.
-Added back in actionCompleted for the dispatch demo when the page loads.
-Added back in actionCompleted for the multi-user collaboration demo.

oliverfcarson committed:
Merge branch 'main' of https://github.com/PubNubDevelopers/healthcare-javascript-demo into devx-4428-vehicle-dispatch
